includes/Plugin.php

```diff
@@ -144,52 +164,64 @@ public static function update() {
                 $options['currency'] = 'rial';
-                update_option( "idpay_cf7_options", $options );
+                update_option("idpay_cf7_options", $options);
             }
-            update_option( 'idpay_cf7_version', '2.1.4' );
+            update_option('idpay_cf7_version', '2.1.4');
 
             //handle the mistake from version 2.1.0
-            if ( file_exists( ABSPATH . "wp-config.php" ) && is_writable( ABSPATH . "wp-config.php" ) ) {
+            if (file_exists(ABSPATH . "wp-config.php") && is_writable(ABSPATH . "wp-config.php")) {
                 self::wp_config_put();
-            } else if ( file_exists( dirname( ABSPATH ) . "/wp-config.php" ) && is_writable( dirname( ABSPATH ) . "/wp-config.php" ) ) {
-                self::wp_config_put( '/' );
+            } else if (file_exists(dirname(ABSPATH) . "/wp-config.php") && is_writable(dirname(ABSPATH) . "/wp-config.php")) {
+                self::wp_config_put('/');
             } else {
                 ?>
                 <div class="error">
-                    <p><?php _e( 'wp-config.php is not writable, please make wp-config.php writable - set it to 0777 temporarily, then set back to its original setting after this plugin has been activated.', 'idpay-contact-form-7' ); ?></p>
+                    <p><?php _e('wp-config.php is not writable, please make wp-config.php writable - set it to 0777 temporarily, then set back to its original setting after this plugin has been activated.', 'idpay-contact-form-7'); ?></p>
                 </div>
                 <?php
                 exit;
             }
 
             //update all the previous tags to new one we defined
-            $rows = $wpdb->get_results( "SELECT * FROM " . $wpdb->prefix . "postmeta WHERE meta_key='_form'" );
-            if ( ! empty( $rows ) ) {
+            $rows = $wpdb->get_results("SELECT * FROM " . $wpdb->prefix . "postmeta WHERE meta_key='_form'");
+            if (!empty($rows)) {
                 foreach ($rows as $row) {
                     $meta_value = preg_replace('/(\[(text))(  *)(idpay_amount){1}(?!\-)(?!\_)(?![A-Za-z_0-9])/', '[payment idpay_amount', $row->meta_value);
                     $meta_value = preg_replace('/(\[(text\*))(  *)(idpay_amount){1}(?!\-)(?!\_)(?![A-Za-z_0-9])/', '[payment* idpay_amount', $meta_value);
-                    $wpdb->update( $wpdb->prefix . 'postmeta',
-                        array( 'meta_value' => $meta_value ),
-                        array( 'meta_id' => $row->meta_id ),
-                        array( '%s' ),
-                        array( '%d' )
+                    $wpdb->update($wpdb->prefix . 'postmeta',
+                        array('meta_value' => $meta_value),
+                        array('meta_id' => $row->meta_id),
+                        array('%s'),
+                        array('%d')
                     );
                 }
             }
         }
+        if(version_compare($version, '2.2.0') < 0)
```

Comment:
Did you even test it???
the version you are checking should be the previous one